### PR TITLE
Fixed extensions import from Chromium based browsers

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/27770

The issue is that Chrome removed extensions list from `Secure Preferences` but some of the browsers still using it. Added extensions list fetching from both places `Secure Preferences` and `Preferences` files to support both ways to keep compatibility with all Chromium based browsers. 
Additionally refactored the import utils to remove existing code duplication to fetch extensions list from external importer and importer list. It allowed to add more tests to import utils.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
- Install latest Chrome (for win I use 109.0.5414.75 (Official Build) (64-bit)) and install an extension from web store, I use jfbnmfgkohlfclfnplnlenbalpppohkm for testing
- Check `c:\Users\[username]\AppData\Local\Google\Chrome\User Data\Default\Secure Preferences` file, it should have one line without extensions info.
- Try to import from Chrome profile
- Make sure installed extension imported to Brave
- Additionally check extensions can be imported from Chrome and others Chromium based browsers who keeps information inside `Secure Preferences`, I checked with Yandex.

### CODE REVIEW

File: common/importer/chrome_importer_utils_unittest.cc [hunk 1]
```diff
@@ -6,56 +6,84 @@
...
+        "Default");
+  }
+  void CopyTestFileToProfile(const std::string& source,
+                             const std::string& target) {
+    base::FilePath test_data_dir;
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+    base::CopyFile(test_data_dir.AppendASCII("import")
+                       .AppendASCII("chrome")
+                       .AppendASCII("default")
+                       .AppendASCII(source),
```
Comment (simonhong) [inline common/importer/chrome_importer_utils_unittest.cc:36]: Should append `source`? below test preferences are located under `test/data/import/chrome/default/`

**Comment by simonhong** [comment]
Do you know upstream changes that causes this failure?
I tried to import from latest canary with our nightly on macOS and it works.

File: common/importer/chrome_importer_utils_unittest.cc (hunk 1 above)
Comment (spylogsster) [reply common/importer/chrome_importer_utils_unittest.cc:36]: it doesnt matter actually, we just need a separate folder for files

**Comment by spylogsster** [comment]
> Do you know upstream changes that causes this failure? I tried to import from latest canary with our nightly on macOS and it works.

it depeds on chromium version, win with Chrome 108 and 109 doesnt work for me

File: common/importer/chrome_importer_utils_unittest.cc (hunk 1 above)
Comment (simonhong) [reply common/importer/chrome_importer_utils_unittest.cc:36]: Oops! it was `source` not `"source"` NVM

**Review by simonhong — APPROVED** [review]
++

**Pull request merged**